docs/index.md

```diff
@@ -125,6 +125,51 @@ provider "confluent" {
 
 !> **Warning:** Hardcoding credentials into a Terraform configuration is not recommended. Hardcoded credentials increase the risk of accidentally publishing secrets to public repositories.
 
+### OAuth Credentials
+
+Confluent Terraform provider allows authentication by using OAuth credentials. You can use the `oauth` block to configure the provider with OAuth credentials.
+
+```shell
+# Option #1: Provide Identity Provider client id and secret, token retrieval URL, and the established Identity Pool ID
+provider "confluent" {
+  cloud_api_key = var.confluent_cloud_api_key
+  cloud_api_secret = var.confluent_cloud_api_secret
+  oauth {
+    oauth_external_token_url = var.oauth_external_token_url            # the URL to retrieve the token from your Identity Provider, such as "https://mycompany.okta.com/oauth2/abc123/v1/token"
+    oauth_external_client_id  = var.oauth_external_client_id           # the client id of your Identity Provider authorization server
+    oauth_external_client_secret = var.oauth_external_client_secret    # the client secret of your Identity Provider authorization server
+    oauth_identity_pool_id = var.oauth_identity_pool_id                # the established Identity Pool ID on Confluent Cloud based on your Identity Provider
+  }
+}
+# Token refresh capability is supported by Confluent Provider for Option #1.
+
+# Option #2: Provide a static token from the Identity Provider the established Identity Pool ID
+provider "confluent" {
+  cloud_api_key = var.confluent_cloud_api_key
+  cloud_api_secret = var.confluent_cloud_api_secret
```

Comment:
nit: can we remove them here and above, but update our note to say you need to add them for managing X, Y, Z resources?